### PR TITLE
WEBUI-2251: Resolve the error-handling and control-flow Sonar findings [maintenance-3.1.x]

### PR DESCRIPTION
**Pull request by vivekchaturved013**
## Problem

SonarCloud reports the error-handling and control-flow subset of the `nuxeo-web-ui` maintainability
backlog: findings that Sonar classes as maintainability but that are the ones most likely to be
hiding a real defect. Jira: [WEBUI-2251](https://hyland.atlassian.net/browse/WEBUI-2251)

`javascript:S2486` x4, `javascript:S4043` x4, `javascript:S1121` x5, `javascript:S6836` x3,
`javascript:S128` x1 (**BLOCKER**), `javascript:S3800` x1, `javascript:S7718` x1.

## Root cause

Each finding needed a judgement about what the code is supposed to do rather than a substitution,
so the reasoning is recorded below per rule.

### S2486 — exceptions should not be ignored (4)

The rule fires when a bound `catch` parameter is never read, so the right fix depends on whether the
throw is expected. **Triage of every site:**

| Site | Verdict | Fix |
| --- | --- | --- |
| `nuxeo-results.js` `items` | **Expected.** Unsafe read of `view.$.list.items` during attach/refresh; already documented and already degrades to `[]`. | Drop the unused binding, keep the comment. |
| `nuxeo-results.js` `_parsePrefMapValue` | **Expected.** Parses a stored preference value, which is user data and may predate the current format. | Drop the unused binding, add a comment. |
| `nuxeo-results.js` `_getDocPrefsFromEnricher` | **Expected.** Same stored-preference parse, from the `userPreferences` enricher. | Drop the unused binding, add a comment. |
| `nuxeo-results.js` `_loadGlobalPrefs` | **Can hide a real failure.** Wraps the `/me/preferences` fetch; on failure the user silently loses their saved results layout for the session. | `console.warn` with the provider and error, keep the graceful fallback to defaults. |
| `nuxeo-document-create.js` `_getSortedSubtypes` | **Can hide a real failure.** Throws when a subtype has no `id`, so the configured `document_type.order` silently stops applying. | `console.warn` with the order config and error, keep the unordered fallback. |

Neither of the two real ones stalls the UI or empties a result list — both already degrade
gracefully and now say so in the console — so **no separate Bug is raised**.

### S4043 — misleading in-place array mutation (4)

All four receivers are arrays built locally in the same function, so no caller shares the reference
and hoisting the call to its own statement is enough; copying would be pointless churn.

| Site | Array | Shared? |
| --- | --- | --- |
| `nuxeo-document-import.js:1313` | `localIndexes` (`const localIndexes = []` in `_processFilesWithMetadata`) | No |
| `nuxeo-document-import.js:1318` | `remoteIndexes` (same function) | No |
| `nuxeo-document-creation-behavior.js:138` | `filteredSubtypes` (built by the `forEach` just above) | No |
| `nuxeo-document-create-shortcuts.js:89` | `shorcuts` (built by the `forEach` just above) | No |

Polymer notification is unaffected: `filteredSubtypes` is sorted **before** `this.set('subtypes', …)`,
so Polymer still sees the assignment; `shorcuts` is never a Polymer property, it is a plain array of
elements handed to `_putNodes`.

### S128 — switch case must end unconditionally (BLOCKER)

`addons/nuxeo-spreadsheet/app/lib/select2-editor.js:95`. The fall-through **was** intentional, but
the guard it fell out of was redundant: `if (ctrlDown) { stopImmediatePropagation(); break; }` and
the `HOME`/`END` case it fell into both called `stopImmediatePropagation()` and broke. A, X, C, V,
home and end now share one case body, which removes the fall-through without changing which keys are
stopped. No keyboard behaviour changes.

### S6836 / S1121 / S7718

Mechanical, but read in context first: brace the `case` bodies that declare `const`/`let` (no name
collides across cases, so nothing was relying on the shared scope); extract the assignments out of
the enclosing call, preserving evaluation order; rename the promise `catch` parameter to `error`.

### S3800 — function should always return the same type

`nuxeo-dropzone._getFiles` returned an array or a single object depending on `multiple`/`blobList`,
and its only caller, `importBatch`, branches on the same flags. Split into `_getFileList` and
`_getFile`, each with a single return type, and let the caller choose. The payloads are byte-for-byte
unchanged.

### Not in this PR

`javascript:S7732` (`docs instanceof Array` in `nuxeo-clipboard.js:258`) is already fixed on the open
WEBUI-2228 branch (#3472 / #3473), so it is not duplicated here.

## Changes

* **`elements/nuxeo-results/nuxeo-results.js`**: three expected catches lose their unused binding and
  gain/keep a comment; `_loadGlobalPrefs` logs the failure it was swallowing.
* **`elements/document/nuxeo-document-create.js`**: `_getSortedSubtypes` logs the malformed-subtype
  failure it was swallowing.
* **`elements/document/nuxeo-document-import.js`**, **`elements/nuxeo-document-creation/nuxeo-document-creation-behavior.js`**,
  **`elements/nuxeo-document-create-actions/nuxeo-document-create-shortcuts.js`**: `sort`/`reverse`
  hoisted out of the enclosing expression.
* **`elements/nuxeo-dropzone/nuxeo-dropzone.js`**: `_getFiles` split into `_getFileList` / `_getFile`.
* **`addons/nuxeo-spreadsheet/app/lib/select2-editor.js`**: braced `ENTER` case body, merged the
  A/X/C/V and HOME/END cases.
* **`addons/nuxeo-spreadsheet/app/ui/spreadsheet.js`**: braced the `userManagerResolver` case body.
* **`addons/nuxeo-spreadsheet/app/utils.js`**, **`app/ui/editors/{document,user,directory}.js`**:
  assignments extracted from the enclosing expressions.
* **`addons/nuxeo-template-rendering/elements/nuxeo-render-template-button.js`**: `catch` parameter
  renamed `response` -> `error`.

## Test plan

- [x] `npm run lint` (eslint) passes; prettier is clean on every changed file
- [x] `npm test` passes — 2769 passed, 1 failed. The single failure,
      `nuxeo-search-form > gives the saved searches dropdown an always visible label (WEBUI-489)`,
      reproduces identically on the untouched base branch (2765 passed, 1 failed), so it is
      pre-existing and unrelated
- [x] Four new/extended unit tests: the `_loadGlobalPrefs` failure path, the `_getSortedSubtypes`
      failure path, the subtype sort order in `_parentChanged`, the shortcut reverse order, and the
      list-vs-object payload choice in `importBatch`
- [x] The descending splice in `_processFilesWithMetadata` is already pinned by the existing
      12-file import tests, which assert exactly which files survive
- [ ] Manual spreadsheet check (not unit-testable — `app/lib/**` and `app/ui/editors/**` patch the
      Handsontable prototype at import time and are excluded from coverage): in a spreadsheet cell
      editor, confirm CTRL+A/C/V/X and plain A/C/V/X, home and end still stay inside the editor, and
      that user/directory/document suggestion cells still render their cached labels

## Notes

Backport pair — the same commit is opened against `lts-2025` and `maintenance-3.1.x`.

Sibling Sonar PRs touch neighbouring lines in some of these files (WEBUI-2248 in
`nuxeo-document-create-shortcuts.js` and `nuxeo-results.js`, WEBUI-2250 and WEBUI-2254 in
`select2-editor.js`); none of them fixes a finding in this ticket's scope, so whichever merges second
may need a small rebase.

[WEBUI-2251]: https://hyland.atlassian.net/browse/WEBUI-2251?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: elements/nuxeo-results/nuxeo-results.js [hunk 1]
```diff
@@ -1452,7 +1453,12 @@ Polymer({
       const parsed = this._parsePrefMapValue(prefsMap[providerName]);
       __globalPrefsCache.set(cacheKey, parsed);
       this.globalPrefs = parsed;
-    } catch (e) {
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.warn('Failed to load global results preferences, falling back to defaults', {
+        provider: providerName,
+        error,
```
[the remark this answers is not in the document]
Comment (vivekchaturved013) [reply elements/nuxeo-results/nuxeo-results.js:1460]: Confirmed, and thanks — the analysis is correct. `_getAllGlobalPreferences()` catches the `preferences.get()` rejection and resolves `{}`, so a backend failure never reaches this catch, and the reset of `__allGlobalPrefsPromise` in `_getAllGlobalPreferencesOnce()` is unreachable for the same reason. One transient failure therefore caches an empty result for the whole session.

Not fixed here, deliberately. Both functions were introduced together in `6ee5d4c6c` (WEBUI-1759, #2995, April 2026) and are untouched by this PR — the diff to this file is only the four `catch` blocks. Propagating the rejection is a runtime behaviour change, and this ticket is scoped to maintainability findings with no behaviour change, so it is raised separately as **WEBUI-2291** (Low) with the root cause, repro and your suggested fix.

Severity is low in practice: `_applyGlobalPrefs()` falls back to the localStorage copy written on every successful save, so a user on their usual browser normally still sees their own layout.

This catch is not dead code and is worth keeping as-is for now. `_configureAllGlobalPreferencesResource()` and the `contentType` assignment run *outside* the inner try, so a throw raised before the request is sent — for example `$.preferences` missing on a detached element — does reject and does reach here. The S2486 finding is resolved either way: the empty catch is gone and the handler is meaningful for that path.

The triage comment on WEBUI-2251 has been corrected accordingly — I had originally described this catch as hiding a silent failure, which was wrong, since the fetch failure is already logged one level down.

File: test/nuxeo-results.test.js [hunk 2]
```diff
@@ -1753,6 +1753,20 @@ suite('nuxeo-results', () => {
       allPrefsStub.restore();
     });
 
+    test('_loadGlobalPrefs falls back to defaults and warns when the request fails', async () => {
+      const allPrefsStub = sinon.stub(results, '_getAllGlobalPreferencesOnce').rejects(new Error('backend down'));
```
[the remark this answers is not in the document]
Comment (vivekchaturved013) [reply test/nuxeo-results.test.js]: Addressed in `97e78239a` / `291e8089a`, taking the second option you offered — describing this as an unexpected helper rejection rather than claiming the request-failure path is covered.

The test is now named `_loadGlobalPrefs falls back to defaults and warns when the preferences load rejects`, with a comment stating that `_getAllGlobalPreferences` swallows a failed request (WEBUI-1759) and that what still reaches this handler is a throw raised before the request is sent, since `_configureAllGlobalPreferencesResource` runs outside that inner catch.

I also added a companion test, `_getAllGlobalPreferences logs and resolves to an empty map when the request fails`, which rejects `preferences.get()` directly. It pins where the failure is actually handled, so the swallow is visible in the suite rather than implied.

I did try the first option — driving `_loadGlobalPrefs` end to end with a rejected `preferences.get()` — but it has to clear the module-level `__allGlobalPrefsPromise` to force its own fetch, and that pushed a later test onto the network where the real request raced its assertions and broke `saveGlobalResultsPrefs persists payload and updates state`. Testing `_getAllGlobalPreferences` directly covers the same behaviour without touching shared state. It will need updating when WEBUI-2291 changes the swallow, and that ticket says so.

**Review by madhurkulshrestha-hyland — APPROVED** [review]

**Review by narasimhahyland — APPROVED** [review]
